wandb/hashutil.py

```diff
@@ -0,0 +1,47 @@
+import base64
+import hashlib
+from typing import Any, NewType, Union
+
+ETag = NewType("ETag", str)
+HexMD5 = NewType("HexMD5", str)
+B64MD5 = NewType("B64MD5", str)
+
+
+def md5_string(string: str) -> B64MD5:
+    return _b64_from_hasher(hashlib.md5(string.encode()))
+
+
+def md5_file_b64(*paths: str) -> B64MD5:
+    return _b64_from_hasher(_md5_file_hasher(*paths))
+
+
+def md5_file_hex(*paths: str) -> HexMD5:
+    return HexMD5(_md5_file_hasher(*paths).hexdigest())
+
+
+def b64_string_to_hex(string: str) -> HexMD5:
+    return HexMD5(base64.standard_b64decode(string).hex())
+
+
+def b64_to_hex_id(id_string: Any) -> str:
+    # TODO: Identify and fix the type of id_string
```

Comment:
what do you want to do about this todo?

Reply:
I'm not actually positive what type this corresponds to. I think it's usually, but not always a `B64MD5`, which I think means we need to keep it a generic `str`.

Reply:
Based on what we do with it, it needs to be a `B64MD5`, so I changed it to that and altered some of the call sites, even though it's clear that's not always correct. I'm going to file a ticket for it actually, because I'm a little worried we might end up in situations where we treat a URI string as b64-encoded data and then crash when we try to decode it.